lib/rx/client.rb

```diff
@@ -18,6 +19,12 @@ class Client < Common::Client::Base
 
     CACHE_TTL = 3600 * 1 # 1 hour cache
 
+    def request(method, path, params = {}, headers = {}, options = {})
```

Comment:
I don't love having to override the request method. There may be a better place to do this, but in the base class the errors are caught in request so I tried to keep the same pattern.

Reply:
Could this be moved into a new  `on_complete` method of the new `rx_raise_error` middleware similiarly to how errors are raised from the original `raise_error` middleware?

Reply:
I don't think so because I don't think we hit the on_complete on timeout errors. But I do think there are probably other places for this. I'll keep playing around with it.

Reply:
oh got it. I think this solution is good then.